application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/views/stacktrace/StacktraceView.java

```diff
@@ -1071,6 +1096,16 @@ public Color getBackground(Object element) {
 		}
 	};
 
+	private static boolean isDarkMode() {
+		Color color = PlatformUI.getWorkbench().getThemeManager().getCurrentTheme().getColorRegistry()
+				.get(JFacePreferences.CONTENT_ASSIST_BACKGROUND_COLOR);
+		if (color.getRed() < 255) {
```

Comment:
Just checking the red component seems a bit specific to a specific theme. Better to instead look at the perceived brightness of the background. Perhaps add another class, since determining this seems like a useful thing to be able to do generally in our UI. Something along the lines of:

import org.eclipse.jface.preference.JFacePreferences;
import org.eclipse.jface.resource.ColorRegistry;
import org.eclipse.swt.graphics.RGB;
import org.eclipse.ui.PlatformUI;

public class ThemeUtils {
    private static final double BRIGHTNESS_THRESHOLD = 0.5;

    public static boolean isDarkMode() {
        ColorRegistry colorRegistry = PlatformUI.getWorkbench().getThemeManager().getCurrentTheme().getColorRegistry();
        RGB backgroundColor = colorRegistry.getRGB(JFacePreferences.CONTENT_ASSIST_BACKGROUND_COLOR);

        if (backgroundColor == null) {
            // Fallback to a default behavior if the color is not available
            return false;
        }

        return calculateBrightness(backgroundColor) < BRIGHTNESS_THRESHOLD;
    }

    private static double calculateBrightness(RGB color) {
        // Using the HSP color model for perceived brightness
        // See: http://alienryderflex.com/hsp.html
        return Math.sqrt(
            0.299 * color.red * color.red +
            0.587 * color.green * color.green +
            0.114 * color.blue * color.blue
        ) / 255.0;
    }
}

Reply:
Sounds good. I will implement this and update the PR.